src/EnergyPlus/HeatBalanceAirManager.cc

```diff
@@ -4902,6 +4908,53 @@ void ReportZoneMeanAirTemp(EnergyPlusData &state)
     // PURPOSE OF THIS SUBROUTINE:
     // This subroutine updates the report variables for the AirHeatBalance.
 
+    if (state.dataHeatBalAirMgr->CalcExtraReportVarMyOneTimeFlag) {
+        for (auto const *reqVar : state.dataOutputProcessor->reqVars) {
+            if (reqVar->name == "ZONE WETBULB GLOBE TEMPERATURE" || reqVar->name == "SPACE WETBULB GLOBE TEMPERATURE") {
+                if (reqVar->key.empty()) {
+                    for (int ZoneLoop = 1; ZoneLoop <= state.dataGlobal->NumOfZones; ++ZoneLoop) {
+                        if (state.dataHeatBal->doSpaceHeatBalanceSimulation) {
+                            for (int spaceNum : state.dataHeatBal->Zone(ZoneLoop).spaceIndexes) {
+                                auto &thisSpaceAirRpt = state.dataHeatBal->spaceAirRpt(spaceNum);
+                                thisSpaceAirRpt.ReportWBGT = true;
+                            }
+                        }
+                        auto &thisZnAirRpt = state.dataHeatBal->ZnAirRpt(ZoneLoop);
+                        thisZnAirRpt.ReportWBGT = true;
+                    }
+                } else {
+                    if (state.dataHeatBal->doSpaceHeatBalanceSimulation) {
+                        int spaceNum = Util::FindItemInList(Util::makeUPPER(reqVar->key), state.dataHeatBal->space);
+                        if (spaceNum > 0) {
+                            auto &thisSpaceAirRpt = state.dataHeatBal->spaceAirRpt(spaceNum);
+                            thisSpaceAirRpt.ReportWBGT = true;
+                        }
+                    }
+                    int ZoneLoop = Util::FindItemInList(Util::makeUPPER(reqVar->key), state.dataHeatBal->Zone);
+                    if (ZoneLoop > 0) {
+                        auto &thisZnAirRpt = state.dataHeatBal->ZnAirRpt(ZoneLoop);
+                        thisZnAirRpt.ReportWBGT = true;
+                    }
+                }
+            }
+        }
+
+        // EMS sensor request WBGT check
+        for (int loop = 1; loop <= state.dataRuntimeLang->NumSensors; ++loop) {
+            if (state.dataRuntimeLang->Sensor(loop).OutputVarName == "ZONE WETBULB GLOBE TEMPERATURE" ||
+                state.dataRuntimeLang->Sensor(loop).OutputVarName == "SPACE WETBULB GLOBE TEMPERATURE") {
```

Comment:
Similar comment here. If Zone then search for matching zone name, if Space, then search for matching space name.

Reply:
I will change it to separate zone and space checks like the pattern you showed above

Reply:
Just to make sure, I don't need to check `state.dataRuntimeLang->Sensor(loop).UniqueKeyName.empty()` right? They're guaranteed to be non-empty (otherwise it will err with "Unique Key Name not found.") @mjwitte

Reply:
> Just to make sure, I don't need to check `state.dataRuntimeLang->Sensor(loop).UniqueKeyName.empty()` right? They're guaranteed to be non-empty (otherwise it will err with "Unique Key Name not found.") @mjwitte

Correct.